File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.arr = []
        curr = 0
        for n in nums:
            curr += n
            self.arr.append(curr)
        

    def sumRange(self, left: int, right: int) -> int:
        rsum = self.arr[right]
        lsum = self.arr[left-1] if left > 0 else 0
        return rsum - lsum
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.java
class NumArray {

    public static int[] prefixSum;
    public NumArray(int[] nums) {
        for(int i =0; i<nums.length; i++){
            if(i-1 >= 0){
                nums[i] += nums[i-1];
            }          
        }
        prefixSum = nums;
    }
    
    public int sumRange(int left, int right) {
        if (left - 1 >= 0){
            return prefixSum[right] - prefixSum[left-1];
        }
        return prefixSum[right];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */